lib/os_files.c

```diff
@@ -1568,21 +1568,23 @@ ___SCMOBJ ___os_executable_path ___PVOID
 #if defined (USE_sysctl) && defined (CTL_KERN) && defined (KERN_PROC) && defined (KERN_PROC_PATHNAME)
 
   {
-    int mib[4] =
+    // Each row has the format: nb_args, arg0, ..., argn
+    int mibs[] = {
 #if defined(KERN_PROC_ARGS)
-      { CTL_KERN, KERN_PROC_ARGS, -1, KERN_PROC_PATHNAME };
-#else
-      { CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1 };
+      4, CTL_KERN, KERN_PROC_ARGS, -1, KERN_PROC_PATHNAME,
 #endif
+      4, CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1,
+      0
+    };
+    int *mibs_iter = (int*)mibs;
```

Comment:
use mibs_probe instead of mibs_iter, and remover (int*) cast which is redundant.